Remove Feature Flag from tasks in Status site.

VinayakaHegade opened the pull request:
<!-- Date Here in dd mmm yyyy-->
Date:  26-01-2024
<!--Developer Name Here-->
Developer Name: @VinayakaHegade

---

## Issue Ticket Number
<!--Issue ticket this PR closes-->
closes #1081

## Description

<!--Description of the changes made in this PR-->
Removed feature flag from task detail page in status site.
https://status.realdevsquad.com/tasks/zkvbxo39CByKmND1mx4Y
### Documentation Updated?

- [ ] Yes
- [x] No

<!--Additional notes about documentation update if applicable-->

### Under Feature Flag

- [ ] Yes
- [ ] No

<!--Indicate if changes are under a feature flag-->

### Database Changes

- [ ] Yes
- [x] No

<!--Notes on any database changes-->

### Breaking Changes

- [ ] Yes
- [x] No

<!--Notes on breaking changes or related issue tickets if applicable-->

### Development Tested?

- [x] Yes
- [ ] No

<!--Confirmation of local testing during development-->

## Screenshots
<details>
<summary>Before</summary>

<!-- Attach your screenshots here👇-->
![before](https://github.com/Real-Dev-Squad/website-status/assets/88454618/023e8bab-432f-4dc8-bcde-837c6a410662)

</details>

<details>
<summary>After</summary>

<!-- Attach your screenshots here👇-->
![after](https://github.com/Real-Dev-Squad/website-status/assets/88454618/7e35b471-ece1-4e42-860e-68985cac3e5e)

</details>

<!--Attach or link to relevant screenshots or visual aids-->

## Test Coverage
<details>
<summary>Screenshot 1</summary>

<!-- Attach your screenshots here👇-->
![coverage](https://github.com/Real-Dev-Squad/website-status/assets/88454618/096e10ee-07d9-477c-aa46-8ce7e58977f4)

</details>

<!--Attach Details on test coverage and outcomes-->

## Additional Notes

<!--Any additional notes, considerations or explanations for reviewers-->

ajoykumardas12 reviewed (commented):
Please provide context on import order changes.

Also please provide context on changed tests because the tests didn't seem related to feature flag.

VinayakaHegade commented:
@ajoykumardas12 . I think import order might be changed because of some of the vs code extensions. And can you let me know what should be the import order that needs to maintained.

And regarding test changes , as i removed the dev flag related code currently changed tests were failing. So removed the unnecessary tests and modified the needed one.

Main changes in tests are because of removal of this piece of code in ProgressDetails component.
`                <li
                    onClick={openDetails}
                    className={styles['list-item']}
                    data-testid="progress-item"
                >
                    {getDateFromTimestamp(data.date)}
                </li>`

`data-testid="progress-item"` was used for querying in ProgressDetails.test.tsx and ProgressCard.test.tsx. So it is the reason for test changes in those two files.

And modal related tests are deleted in  ProgressDetails.test.tsx because, after the dev flag removal `showProgressDetails` state would never become true and modal will not open, so modal tests are deleted.

ajoykumardas12 commented:
> I think import order might be changed because of some of the vs code extensions. And can you let me know what should be the import order that needs to maintained.

I don't think status site has specified import order to follow, unrelated changes to import would be unnecessary though. But, if you want to follow import order, I think first react,nextjs imports, then other libraries, then components, utils, then styles, icons would be good.

Achintya-Chatterjee commented:
can you please add a working proof after removing the feature flag

Achintya-Chatterjee reviewed (commented):
Can you please add a working proof after removing the feature flag

VinayakaHegade commented:
> Can you please add a working proof after removing the feature flag

Hey @Achintya-Chatterjee ,
I have added screen recording of working demo in the screenshots section above.
Can you review it again and check if it can be approved.

Ajeyakrishna-k commented:
In new progress updates view, there was an issue where we were not able to view all details(like Next update, Blockers) for old updates. Is that fixed? 

cc: @satyam73

VinayakaHegade commented:
> In new progress updates view, there was an issue where we were not able to view all details(like Next update, Blockers) for old updates. Is that fixed?
> 
> cc: @satyam73

Details like Next update, blockers were earlier shown in modal. But the new progress update view doesn't have any modal and shows only Title and time in days ago format for old updates.

This PR doesn’t intend to fix any bugs. It’s about removing the dev flag in the taskDetails page as Ankush had asked.

If this needs to be put on hold until the bug is fixed, please let me know @Ajeyakrishna-k.

Ajeyakrishna-k commented:
> > In new progress updates view, there was an issue where we were not able to view all details(like Next update, Blockers) for old updates. Is that fixed?
> > cc: @satyam73
> 
> Details like Next update, blockers were earlier shown in modal. But the new progress update view doesn't have any modal and shows only Title and time in days ago format for old updates.
> 
> This PR doesn’t intend to fix any bugs. It’s about removing the dev flag in the taskDetails page as Ankush had asked.
> 
> If this needs to be put on hold until the bug is fixed, please let me know @Ajeyakrishna-k.

Yep, please have a look at these messages:

https://discord.com/channels/673083527624916993/741616265050849282/1194286232071573636
https://discord.com/channels/673083527624916993/741616265050849282/1195018552428736642

satyam73 commented:
> In new progress updates view, there was an issue where we were not able to view all details(like Next update, Blockers) for old updates. Is that fixed?
> 
> cc: @satyam73

Not now I had previously raised PR in which modal can be opened then Ankush told me to expand the card. 

The PR for modal is here (which is not the requirement now) - https://github.com/Real-Dev-Squad/website-status/pull/1075

I haven't worked on expanding the card part.
